test_conformance/extensions/cl_khr_command_buffer/negative_command_buffer_barrier.cpp

```diff
@@ -0,0 +1,199 @@
+// command-buffer.
+struct CommandBufferBarrierInvalidCommandBuffer : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clCommandBarrierWithWaitListKHR(
+            nullptr, queue, 0, nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_COMMAND_BUFFER_KHR,
+                               "clCommandBarrierWithWaitListKHR should return "
+                               "CL_INVALID_COMMAND_BUFFER_KHR",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+
+// CL_INVALID_OPERATION if command_buffer has been finalized.
+struct CommandBufferBarrierBufferFinalized : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_int error = clFinalizeCommandBufferKHR(command_buffer);
+        test_error(error, "clFinalizeCommandBufferKHR failed");
+
+        error = clCommandBarrierWithWaitListKHR(command_buffer, nullptr, 0,
+                                                nullptr, nullptr, nullptr);
+
+        test_failure_error_ret(error, CL_INVALID_OPERATION,
+                               "clCommandBarrierWithWaitListKHR should return "
+                               "CL_INVALID_OPERATION",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+
+// CL_INVALID_VALUE if mutable_handle is not NULL.
+struct CommandBufferBarrierMutableHandleNotNull : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_mutable_command_khr mutable_handle;
+
+        cl_int error = clCommandBarrierWithWaitListKHR(
+            command_buffer, nullptr, 0, nullptr, nullptr, &mutable_handle);
+
+        test_failure_error_ret(error, CL_INVALID_VALUE,
+                               "clCommandBarrierWithWaitListKHR should return "
+                               "CL_INVALID_VALUE",
+                               TEST_FAIL);
+
+        return CL_SUCCESS;
+    }
+};
+
+// CL_INVALID_SYNC_POINT_WAIT_LIST_KHR if sync_point_wait_list is NULL and
+// num_sync_points_in_wait_list is > 0, or sync_point_wait_list is not NULL and
+// num_sync_points_in_wait_list is 0, or if synchronization-point objects in
+// sync_point_wait_list are not valid synchronization-points.
+struct CommandBufferBarrierSyncPointsNullOrNumZero
+    : public BasicCommandBufferTest
+{
+    using BasicCommandBufferTest::BasicCommandBufferTest;
+
+    cl_int Run() override
+    {
+        cl_sync_point_khr invalid_point = 0;
+        std::vector<cl_sync_point_khr*> invalid_sync_points;
+        invalid_sync_points.push_back(&invalid_point);
+
+        cl_int error = clCommandBarrierWithWaitListKHR(
+            command_buffer, nullptr, 1, *invalid_sync_points.data(), nullptr,
+            nullptr);
```

Comment:
We don't need a vector here (and below), this could be simplified to
```
        cl_sync_point_khr invalid_point = 0;

        cl_int error = clCommandBarrierWithWaitListKHR(
            command_buffer, nullptr, 1, &invalid_point, nullptr,
            nullptr);
```